test/components/embedders/test_sentence_transformers_doc_image_embedder.py

```diff
@@ -0,0 +1,396 @@
+            device=ComponentDevice.from_str("cpu"),
+            model_kwargs={
+                "file_name": "onnx/model.onnx"
+            },  # setting the path isn't necessary if the repo contains a "onnx/model.onnx" file but this is to prevent a HF warning
+            backend="onnx",
+        )
+        onnx_embedder.warm_up()
+
+        mocked_factory.get_embedding_backend.assert_called_once_with(
+            model="sentence-transformers/all-MiniLM-L6-v2",
+            device="cpu",
+            auth_token=None,
+            trust_remote_code=False,
+            local_files_only=False,
+            model_kwargs={"file_name": "onnx/model.onnx"},
+            tokenizer_kwargs=None,
+            config_kwargs=None,
+            backend="onnx",
+        )
+
+    @patch(
+        "haystack_experimental.components.embedders.sentence_transformers_doc_image_embedder._SentenceTransformersEmbeddingBackendFactory"
+    )
+    def test_model_openvino_backend(self, mocked_factory):
+        openvino_embedder = SentenceTransformersDocumentImageEmbedder(
+            model="sentence-transformers/all-MiniLM-L6-v2",
+            token=None,
+            device=ComponentDevice.from_str("cpu"),
+            model_kwargs={
+                "file_name": "openvino/openvino_model.xml"
+            },  # setting the path isn't necessary if the repo contains a "openvino/openvino_model.xml" file but this is to prevent a HF warning
+            backend="openvino",
+        )
+        openvino_embedder.warm_up()
+
+        mocked_factory.get_embedding_backend.assert_called_once_with(
+            model="sentence-transformers/all-MiniLM-L6-v2",
+            device="cpu",
+            auth_token=None,
+            trust_remote_code=False,
+            local_files_only=False,
+            model_kwargs={"file_name": "openvino/openvino_model.xml"},
+            tokenizer_kwargs=None,
+            config_kwargs=None,
+            backend="openvino",
+        )
+
+    def test_validate_image_paths(self, test_files_path):
+        embedder = SentenceTransformersDocumentImageEmbedder(model="model")
+
+        image_paths = glob.glob(str(test_files_path / "images" / "*.*")) + glob.glob(str(test_files_path / "pdf" / "*.pdf"))
+
+        documents = []
+        for i, path in enumerate(image_paths):
+            document = Document(content=f"document number {i}", meta={"file_path": path})
+            if path.endswith(".pdf"):
+                document.meta["page_number"] = 1
+            documents.append(document)
+
+        validated_paths = embedder._validate_image_paths(documents)
+        assert len(validated_paths) == len(documents)
+        for path_info in validated_paths:
+            assert str(path_info["path"]) in image_paths
+            assert path_info["type"] in ["image", "pdf"]
+            if path_info["type"] == "pdf":
+                assert path_info["page_number"] == 1
+            else:
+                assert "page_number" not in path
```

Comment:
I think this is wrong? `path` is a for loop variable so I think you meant `path_info["path"]`?